src/OSSupport/NetworkSingleton.cpp

```diff
@@ -18,8 +18,27 @@
 
 
 
-cNetworkSingleton::cNetworkSingleton(void):
-	m_HasTerminated(false)
+cNetworkSingleton::~cNetworkSingleton()
+{
+	// Check that Terminate has been called already:
+	ASSERT(m_HasTerminated);
+}
```

Comment:
What about the missing constructor here? This will throw a warning of non-initialized variables